File: src/main/java/ai/devchat/idea/SensitiveDataStorage.java
package ai.devchat.idea;

import com.intellij.credentialStore.CredentialAttributes;
import com.intellij.credentialStore.CredentialAttributesKt;
import com.intellij.credentialStore.Credentials;
import com.intellij.ide.passwordSafe.PasswordSafe;

public class SensitiveDataStorage {

    private static final String KEY_NAME = "USER_KEY";
    private static final String SUBSYSTEM = "ai.devchat.idea";

    private static CredentialAttributes createCredentialAttributes() {
        return new CredentialAttributes(
                CredentialAttributesKt.generateServiceName(SUBSYSTEM, KEY_NAME));
    }

    public static void setKey(String key) {
        CredentialAttributes attributes = createCredentialAttributes();
        Credentials credentials = new Credentials("default", key);
        PasswordSafe.getInstance().set(attributes, credentials);
    }

    public static String getKey() {
        CredentialAttributes attributes = createCredentialAttributes();
        return PasswordSafe.getInstance().getPassword(attributes);
    }
}

File: src/main/java/ai/devchat/idea/storage/SensitiveDataStorage.kt
package ai.devchat.idea.storage

import com.intellij.credentialStore.CredentialAttributes
import com.intellij.credentialStore.Credentials
import com.intellij.credentialStore.generateServiceName
import com.intellij.ide.passwordSafe.PasswordSafe

object SensitiveDataStorage {
    private const val KEY_NAME = "USER_KEY"
    private const val SUBSYSTEM = "ai.devchat.idea"
    private fun createCredentialAttributes(): CredentialAttributes {
        return CredentialAttributes(
            generateServiceName(SUBSYSTEM, KEY_NAME)
        )
    }

    var key: String?
        get() {
            val attributes = createCredentialAttributes()
            return PasswordSafe.instance.getPassword(attributes)
        }
        set(key) {
            val attributes = createCredentialAttributes()
            val credentials = Credentials("default", key)
            PasswordSafe.instance.set(attributes, credentials)
        }
}
